### PR TITLE
Fix: Only pass street name to GovPay if provided

### PR DESCRIPTION
**Pull request by DGajewska**
IWTF-2170

* Adds conditonal logic to prefilled billing address
* Adds tests for payment.js

### CODE REVIEW

File: packages/gafl-webapp-service/src/processors/__tests__/payment.spec.js [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+describe('preparePayment', () => {
+  let request, transaction, result
+  beforeEach(() => {
+    request = createRequest()
+    transaction = createTransaction()
+    result = preparePayment(request, transaction)
+  })
+  describe('provides the correct return url', () => {
+    it('when the "x-forwarded-proto" header is present', () => {
+      expect(result.return_url).toBe('https://localhost:1234/buy/agreed')
```
Comment (jaucourt) [inline packages/gafl-webapp-service/src/processors/__tests__/payment.spec.js:42]: After looking for a bit, I can see that `result.return_url` should be the same as ` ``http://${request.info.host}/buy/agreed`` `, however a bit more of a thorough test would be a bit more descriptive. If you were to modify the `createRequest` method to allow for host to be specified, but default to `localhost:1234`, you could then do something like:
```
['localhost:4321', 'otherhost:8888', 'samplehost:4444'].forEach(host => 
  it(`uses SSL when 'x-forwarded-proto' header is present, host '${host}'`, () => {
    const request = createRequest(host)
    expect(result.return_url).toBe(`https://${host}/buy/agreed`)
  })
)
```

File: packages/gafl-webapp-service/src/processors/__tests__/payment.spec.js [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    })
+    
+    it('when the "x-forwarded-proto" header is not present', () => {
+      request.headers = {}
+      const result = preparePayment(request, transaction)
+      expect(result.return_url).toBe('http://localhost:1234/buy/agreed')
```
Comment (jaucourt) [inline packages/gafl-webapp-service/src/processors/__tests__/payment.spec.js:48]: Same here, but use a different list to previously

**Review by jaucourt — CHANGES_REQUESTED** [review]
It's great that you've added in quite a few new tests that cover more than the new code you've added, thanks for doing that. I've made a couple of suggestions to broaden tests to be a little more expressive, this could be done with most of your tests, as at the moment the tests could be passed by simply hardcoding values. This probably wouldn't ever happen, but it would make them slightly more expressive and easy to understand, as well as allowing edge cases to be tested.

**Review by nabeelamir-defra — APPROVED** [review]

**Review by jaucourt — APPROVED** [review]
